
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Concepts of Community Analysis &#8212; VMI Bootcamp II 1.0 documentation</title>
    <link rel="stylesheet" href="../_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Microbiome Analysis and Hypothesis Testing" href="5_analysis_and_hypothesis_testing.html" />
    <link rel="prev" title="From Sequences to Community Composition" href="3_sequences_to_composition.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="5_analysis_and_hypothesis_testing.html" title="Microbiome Analysis and Hypothesis Testing"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="3_sequences_to_composition.html" title="From Sequences to Community Composition"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">VMI Bootcamp II 1.0 documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="concepts-of-community-analysis">
<h1>Concepts of Community Analysis<a class="headerlink" href="#concepts-of-community-analysis" title="Permalink to this headline">¶</a></h1>
<p><strong>How can we examine microbial communities?</strong></p>
<p><strong>Let’s switch datasets and look at:</strong> <a class="reference external" href="http://humanfoodproject.com/americangut/">The American Gut Project (AGP)</a></p>
<div class="figure" id="id1">
<a class="reference internal image-reference" href="../_images/agp.jpg"><img alt="agp" src="../_images/agp.jpg" style="width: 86.0px; height: 42.8px;" /></a>
<p class="caption"><span class="caption-text">American Gut Project (AGP) logo. Source: <a class="reference external" href="http://humanfoodproject.com/americangut/">http://humanfoodproject.com/americangut/</a></span></p>
</div>
<p><strong>Why?</strong></p>
<blockquote>
<div><ul class="simple">
<li><dl class="first docutils">
<dt>Clustering features, creating a phylogeny, and assigning taxonomy are the most computationally intensive</dt>
<dd><ul class="first last">
<li>We used very small and sparse data</li>
<li>This subset of the AGP has 1,375 individuals</li>
<li>There are 5,590 features (bacteria) and ~200 metadata measurements</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>You may get processed data already as a table, phylogeny…</dt>
<dd><ul class="first last">
<li>Need to know how to import at any stage of the project</li>
</ul>
</dd>
</dl>
</li>
</ul>
</div></blockquote>
<dl class="docutils">
<dt><strong>Importing the AGP</strong></dt>
<dd><ol class="first last arabic simple">
<li>Look at the files in the american_gut folder</li>
<li>Import the biom table</li>
<li>Import the bacterial phylogeny</li>
<li>Look at the available metadata</li>
</ol>
</dd>
</dl>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># See the files we will work with</span>
ls -lsh american_gut

<span class="c1"># Import the biom count table of features by samples</span>
qiime tools import <span class="se">\</span>
  --input-path  american_gut/1_0_biom.biom <span class="se">\</span>
  --type <span class="s1">&#39;FeatureTable[Frequency]&#39;</span> <span class="se">\</span>
  --source-format BIOMV100Format <span class="se">\</span>
  --output-path american_gut/1_1_biom.qza

<span class="c1"># Import the bacterial 16S phylogeny</span>
qiime tools import <span class="se">\</span>
  --input-path american_gut/1_0_tree.tre <span class="se">\</span>
  --output-path american_gut/1_1_tree.qza <span class="se">\</span>
  --type <span class="s1">&#39;Phylogeny[Rooted]&#39;</span>

<span class="c1"># Create a visual of the metadata (map.txt)</span>
qiime metadata tabulate <span class="se">\</span>
  --m-input-file american_gut/1_0_map.txt <span class="se">\</span>
  --o-visualization american_gut/1_1_metadata.qzv

<span class="c1"># Look at metadata available</span>
qiime tools view american_gut/1_1_metadata.qzv
</pre></div>
</div>
<div class="figure" id="id2">
<a class="reference internal image-reference" href="../_images/agp_metadata.png"><img alt="agp_metadata" src="../_images/agp_metadata.png" style="width: 1535.4px; height: 651.0px;" /></a>
<p class="caption"><span class="caption-text">AGP metadata.</span></p>
</div>
<div class="section" id="quality-control">
<h2>Quality Control<a class="headerlink" href="#quality-control" title="Permalink to this headline">¶</a></h2>
<p><strong>What should we do to improve the resolution and remove noise?</strong></p>
<dl class="docutils">
<dt><strong>1. Filter features from the observation table by:</strong></dt>
<dd><ul class="first last simple">
<li>Minimum or maximum frequency (counts across samples)</li>
<li>Minimum or maximum ubiquity (proportion of samples)</li>
<li>Where metadata is equal to a certain value</li>
<li>Removing specific samples by the unqiue id</li>
</ul>
</dd>
</dl>
<p><strong>Options for specifically filtering features:</strong></p>
<blockquote>
<div><table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--i-table <var>ARTIFACT_PATH</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>The feature table from which features should
be filtered.  [required]</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--p-min-frequency <var>INTEGER</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>The minimum total frequency that a feature
must have to be retained.  [default: 0]</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--p-max-frequency <var>INTEGER</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>The maximum total frequency that a feature
can have to be retained. If no value is
provided this will default to infinity
(i.e., no maximum frequency filter will be
applied).  [optional]</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--p-min-samples <var>INTEGER</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>The minimum number of samples that a feature
must be observed in to be retained.
[default: 0]</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--p-max-samples <var>INTEGER</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>The maximum number of samples that a feature
can be observed in to be retained. If no
value is provided this will default to
infinity (i.e., no maximum sample filter
will be applied).  [optional]</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--m-metadata-file <var>MULTIPLE_PATH</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>Metadata file or artifact viewable as
metadata. This option may be supplied
multiple times to merge metadata. Feature
metadata used with <cite>where</cite> parameter when
selecting features to retain, or with
<cite>exclude_ids</cite> when selecting features to
discard.  [optional]</td></tr>
<tr><td class="option-group">
<kbd><span class="option">--p-where <var>TEXT</var></span></kbd></td>
<td>SQLite WHERE clause specifying feature
metadata criteria that must be met to be
included in the filtered feature table. If
not provided, all features in <cite>metadata</cite>
that are also in the feature table will be
retained.  [optional]</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--p-exclude-ids</span></kbd></td>
</tr>
<tr><td>&#160;</td><td>Samples to remove</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--p-no-exclude-ids</span></kbd></td>
</tr>
<tr><td>&#160;</td><td>If true, the features selected by <cite>metadata</cite>
or <cite>where</cite> parameters will be excluded from
the filtered table instead of being
retained.  [default: False]</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--o-filtered-table <var>ARTIFACT_PATH</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>FeatureTable[Frequency] The resulting feature table filtered by feature.  [required if not passing –output-dir]</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--output-dir <var>DIRECTORY</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>Output unspecified results to a directory</td></tr>
</tbody>
</table>
</div></blockquote>
<p><strong>Let’s remove really rare features, to cut down on sequencing artefacts</strong></p>
<p><strong>Let’s also remove features not in at least 20% of samples</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime feature-table filter-features <span class="se">\</span>
   --i-table american_gut/1_1_biom.qza <span class="se">\</span>
   --o-filtered-table american_gut/1_2_biom_min_50.qza <span class="se">\</span>
   --p-min-frequency <span class="m">50</span>

qiime feature-table filter-features <span class="se">\</span>
   --i-table american_gut/1_2_biom_min_50.qza <span class="se">\</span>
   --o-filtered-table american_gut/1_2_biom_min_50_ubiq_20.qza <span class="se">\</span>
   --p-min-samples <span class="m">275</span>

qiime feature-table summarize <span class="se">\</span>
   --i-table american_gut/1_2_biom_min_50_ubiq_20.qza <span class="se">\</span>
   --o-visualization american_gut/1_2_biom_summary.qzv

qiime tools view american_gut/1_2_biom_summary.qzv
</pre></div>
</div>
<p><strong>Let’s see how we did:</strong></p>
<p>Looks like we cut down our features from 5,590 to 664, not bad…</p>
<div class="figure">
<a class="reference internal image-reference" href="../_images/agp_filtered.png"><img alt="agp_filtered" src="../_images/agp_filtered.png" style="width: 1560.3px; height: 562.1px;" /></a>
</div>
<p>A couple features dominate the counts…</p>
<div class="figure">
<a class="reference internal image-reference" href="../_images/agp_feature_counts.png"><img alt="agp_feature_counts" src="../_images/agp_feature_counts.png" style="width: 1100.8px; height: 347.2px;" /></a>
</div>
<dl class="docutils">
<dt>Keep in mind that microbiome counts are <em>not</em> normally distributed</dt>
<dd><ul class="first last simple">
<li>More often follow a Poisson / Negative Binomial distribution</li>
</ul>
</dd>
</dl>
<div class="figure" id="id3">
<a class="reference internal image-reference" href="../_images/poisson.png"><img alt="poisson" src="../_images/poisson.png" style="width: 292.5px; height: 234.0px;" /></a>
<p class="caption"><span class="caption-text">Different Poisson distributions can be modeled, microbiome counts will look like the yellow most often. Source: Wikipedia</span></p>
</div>
<p>We also want to remove samples with low counts, and there is a nice tool to choose that cutoff…</p>
<div class="figure">
<a class="reference internal image-reference" href="../_images/agp_sample_cutoff.png"><img alt="agp_sample_cutoff" src="../_images/agp_sample_cutoff.png" style="width: 402.3px; height: 307.8px;" /></a>
</div>
<dl class="docutils">
<dt><strong>Keep in mind you could also:</strong></dt>
<dd><ul class="first last simple">
<li>Remove particular microbial taxon: –p-exclude Wolbachia</li>
<li>Remove specific samples: –p-exclude-ids samples-to-remove.tsv</li>
<li>Remove samples based on metadata: –p-where “sex IN (‘male’, ‘female’)”</li>
</ul>
</dd>
</dl>
<p><strong>2. We can remove samples with the filter-samples tool</strong></p>
<dl class="docutils">
<dt>Let’s remove samples with less than 50,000 counts</dt>
<dd><ul class="first last simple">
<li>This is a high cutoff, and will cut us down to only 53 samples</li>
<li>This will make the tutorial easier, but 1,000 or 10,000 is often more than adequate</li>
</ul>
</dd>
</dl>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime feature-table filter-samples <span class="se">\</span>
   --i-table american_gut/1_2_biom_min_50_ubiq_20.qza <span class="se">\</span>
   --o-filtered-table american_gut/1_3_biom_samplemin_50k.qza <span class="se">\</span>
   --p-min-frequency <span class="m">50000</span>

qiime feature-table summarize <span class="se">\</span>
   --i-table american_gut/1_3_biom_samplemin_50k.qza <span class="se">\</span>
   --o-visualization american_gut/1_3_biom_summary.qzv

qiime tools view american_gut/1_3_biom_summary.qzv
</pre></div>
</div>
<p>This is nice for our purposes today, but note that some samples still have much higher counts than others…</p>
<div class="figure">
<a class="reference internal image-reference" href="../_images/agp_sam_filter.png"><img alt="agp_sam_filter" src="../_images/agp_sam_filter.png" style="width: 662.0px; height: 458.0px;" /></a>
</div>
<dl class="docutils">
<dt><strong>Let’s work with this cleaner data, while keeping in mind there are tons of other ways to filter!</strong></dt>
<dd><ul class="first last simple">
<li>Remove sequences before Deblur feature analysis: qiime quality-control exclude-seqs</li>
<li>Remove samples and features based on the mock community: qiime quality-control evaluate-composition</li>
<li><a class="reference external" href="https://docs.qiime2.org/2018.8/tutorials/filtering/">More extensive filtering documentation</a>.</li>
</ul>
</dd>
</dl>
</div>
<div class="section" id="community-transformation-and-examination">
<h2>Community Transformation and Examination<a class="headerlink" href="#community-transformation-and-examination" title="Permalink to this headline">¶</a></h2>
<p><strong>Let’s look at the data beyond just a table summary!</strong></p>
<p><strong>The feature-table tool has a lot of options:</strong></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime feature-table --help
</pre></div>
</div>
<dl class="docutils">
<dt>Commands:</dt>
<dd><table class="first last docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">-c<var>ore-features</var></span></kbd></td>
<td>Identify core features in table</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-f<var>ilter-features</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>Filter features from table</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-f<var>ilter-samples</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>Filter samples from table</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-f<var>ilter-seqs</var></span></kbd></td>
<td>Filter features from sequences</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-g<var>roup</var></span></kbd></td>
<td>Group samples or features by a metadata column</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-h<var>eatmap</var></span></kbd></td>
<td>Generate a heatmap representation of a feature table</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-m<var>erge</var></span></kbd></td>
<td>Combine multiple tables</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-m<var>erge-seqs</var></span></kbd></td>
<td>Combine collections of feature sequences</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-m<var>erge-taxa</var></span></kbd></td>
<td>Combine collections of feature taxonomies</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-p<var>resence-absence</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>Convert to presence/absence</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-r<var>arefy</var></span></kbd></td>
<td>Rarefy table</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-r<var>elative-frequency</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>Convert to relative frequencies</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-s<var>ubsample</var></span></kbd></td>
<td>Subsample table</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-s<var>ummarize</var></span></kbd></td>
<td>Summarize table</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-t<var>abulate-seqs</var></span></kbd></td>
<td>View sequence associated with each feature</td></tr>
</tbody>
</table>
</dd>
</dl>
<p>I’m going to create a shortcut variable for the path to our table, since we will always come back to it</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nv">table</span><span class="o">=</span>american_gut/1_3_biom_samplemin_50k.qza
</pre></div>
</div>
<p>At this point in a real analysis I would also back up the raw and filtered tables elsewhere</p>
<p>Note how I use the shortcut to our table instead of writing it out…</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime feature-table heatmap <span class="se">\</span>
   --i-table <span class="s2">&quot;</span><span class="nv">$table</span><span class="s2">&quot;</span> <span class="se">\</span>
   --o-visualization american_gut/2_0_table_heatmap.qzv

qiime tools view american_gut/2_0_table_heatmap.qzv
</pre></div>
</div>
<dl class="docutils">
<dt><strong>How is our community taxonomically distributed?</strong></dt>
<dd><ul class="first last simple">
<li>Let’s examine barplots of the taxonomic distribution for each sample</li>
<li>These plots are based on relative abundance within a total sample</li>
<li>Do not provide information about absolute microbial load</li>
</ul>
</dd>
</dl>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1">### MAKE TAXONOMY BARPLOT ###</span>
qiime taxa barplot <span class="se">\</span>
   --i-table <span class="s2">&quot;</span><span class="nv">$table</span><span class="s2">&quot;</span> <span class="se">\</span>
   --i-taxonomy american_gut/1_1_biom.qza <span class="se">\</span>
   --m-metadata-file american_gut/1_0_map.txt <span class="se">\</span>
   --o-visualization 1_2_taxonomy_barplot.qzv

qiime tools view 1_18_taxonomy_barplot.qzv
</pre></div>
</div>
</div>
<div class="section" id="rarefaction-and-normalization">
<h2>Rarefaction and Normalization<a class="headerlink" href="#rarefaction-and-normalization" title="Permalink to this headline">¶</a></h2>
<dl class="docutils">
<dt><strong>Raw counts are biased:</strong></dt>
<dd><ul class="first last simple">
<li>Samples have different numbers of sequence counts</li>
<li>Not indicative of absolute bacterial abundance</li>
<li>Artefact of extraction, library preparation and pooling, and sequencing efficiency</li>
</ul>
</dd>
</dl>
<p><strong>Some ways to transform the counts:</strong></p>
<blockquote>
<div><table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-p<var>resence-absence</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>Convert to presence/absence</td></tr>
<tr><td class="option-group">
<kbd><span class="option">-r<var>arefy</var></span></kbd></td>
<td>Rarefy table</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">-r<var>elative-frequency</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>Convert to relative frequencies</td></tr>
</tbody>
</table>
</div></blockquote>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>qiime feature-table presence-absence <span class="se">\ </span>...
qiime feature-table rarefy <span class="se">\ </span>...
qiime feature-table relative-frequency <span class="se">\ </span>...
</pre></div>
</div>
<p><strong>Presence / Absence</strong>: value of 1 if taxon is present in sample, 0 otherwise
<strong>Rarefy</strong>: subsample the same number of sequence from each sample
<strong>Relative Frequency</strong>: propotion of each feature out of the total feature counts for each sample</p>
<dl class="docutils">
<dt><strong>There are other normalization methods, often based on RNA sequencing methods.</strong></dt>
<dd><ul class="first last simple">
<li>Generally seem more biased than rarefy / relative abundance</li>
<li>Better fit in standard sense, don’t seem to adapt well to wide bias across different datasets</li>
</ul>
</dd>
</dl>
<p>I’d suggest sticking to relative abundance and rarefied communities!</p>
</div>
<div class="section" id="alpha-and-beta-diversity">
<h2>Alpha and Beta Diversity<a class="headerlink" href="#alpha-and-beta-diversity" title="Permalink to this headline">¶</a></h2>
<p><strong>We can use ecological metrics to understand community composition and inter-sample microbiome similarity!</strong></p>
<p><strong>Alpha Diversity</strong>: Measure of diversity within a sample</p>
<p><strong>Beta Diversity</strong>: Measure of microbiome similarity between a pair of samples</p>
<p><a class="reference external" href="https://forum.qiime2.org/t/alpha-and-beta-diversity-explanations-and-commands/2282">There are many metrics (ways to measure) Alpha and Beta Diversity</a></p>
<p><strong>Both Alpha and Beta Diversity can take into account the microbial phylogeny with certain metrics</strong></p>
<p>Metrics are assigned with the <em>-p-metric</em> argument:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># Alpha Diversity - Non-Phylogenetic Observed OTUs</span>
qiime diversity alpha <span class="se">\</span>
  --i-table table.qza <span class="se">\</span>
  --p-metric observed_otus <span class="se">\</span>
  --o-alpha-diversity observed_otus_vector.qza

<span class="c1"># Alpha Diversity - Non-Phylogenetic Observed OTUs</span>
qiime diversity alpha-phylogenetic <span class="se">\</span>
  --i-table table.qza <span class="se">\</span>
  --i-phylogeny rooted-tree.qza <span class="se">\</span>
  --p-metric faith_pd <span class="se">\</span>
  --o-alpha-diversity faith_pd_vector.qza

<span class="c1"># Beta Diversity - Non-Phylogenetic Bray-Curtis</span>
qiime diversity beta <span class="se">\</span>
  --i-table table.qza <span class="se">\</span>
  --p-metric braycurtis <span class="se">\</span>
  --o-distance-matrix unweighted_unifrac_distance_matrix.qza

<span class="c1"># Beta Diversity - Phylogenetic Unweighted UNIFRAC</span>
qiime diversity beta-phylogenetic <span class="se">\</span>
   --i-table table.qza <span class="se">\</span>
   --i-phylogeny rooted-tree.qza <span class="se">\</span>
   --p-metric unweighted_unifrac <span class="se">\</span>
   --o-distance-matrix unweighted_unifrac_distance_matrix.qza
</pre></div>
</div>
<dl class="docutils">
<dt><strong>QIIME2 has done a good job wrapping diversity computation with my favorite metrics into one easy script:</strong></dt>
<dd><blockquote class="first">
<div><ul class="simple">
<li>Rarefies the table to a specified depth (necessary for diversity to be computed accurately)</li>
<li><dl class="first docutils">
<dt>Calculates Alpha and Beta Diversity</dt>
<dd><ul class="first last">
<li><dl class="first docutils">
<dt>Alpha Metrics:</dt>
<dd><ul class="first last">
<li>Faith’s PD: Phylogenetic diversity</li>
<li>Observed OTUs: Richness of community</li>
<li>Shannon: Balances richness and evenness</li>
<li>Pielou’s Evenness: Evenness of community</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>Beta Diversity</dt>
<dd><ul class="first last">
<li>Unweighted Unifrac: Presence / absence phylogenetic distance between samples</li>
<li>Weighted Unifrac: Abundance weighted phylogenetic distance between samples</li>
<li>Jaccard: Presence / absence distance between samples</li>
<li>Bray Curtis: Abundance weighted distance between samples</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
<li>Can be parallelized across multiple CPUs</li>
<li>Generates output visuals</li>
</ul>
</div></blockquote>
<dl class="last docutils">
<dt>Options:</dt>
<dd><table class="first last docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--i-table <var>ARTIFACT_PATH</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>The feature table containing the samples
over which diversity metrics should be
computed.  [required]</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--i-phylogeny <var>ARTIFACT_PATH</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>Phylogenetic tree containing tip identifiers
that correspond to the feature identifiers
in the table. This tree can contain tip ids
that are not present in the table, but all
feature ids in the table must be present in
this tree.  [required]</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--p-sampling-depth <var>INTEGER_RANGE</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>The total frequency that each sample should
be rarefied to prior to computing diversity
metrics.  [required]</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--m-metadata-file <var>MULTIPLE_PATH</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>Metadata file or artifact viewable as
metadata. This option may be supplied
multiple times to merge metadata. The sample
metadata to use in the emperor plots.
[required]</td></tr>
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--p-n-jobs <var>INTEGER_RANGE</var></span></kbd></td>
</tr>
<tr><td>&#160;</td><td>The number of jobs to use for the computation. This works
by breaking down the pairwise matrix into
n_jobs even slices and computing them in
parallel. If -1 all CPUs are used. If 1 is
given, no parallel computing code is used at
all, which is useful for debugging. For
n_jobs below -1, (n_cpus + 1 + n_jobs) are
used. Thus for n_jobs = -2, all CPUs but one
are used. (Description from
sklearn.metrics.pairwise_distances)
[default: 1]</td></tr>
</tbody>
</table>
</dd>
</dl>
</dd>
</dl>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1">###############################################################################</span>
<span class="c1">##### SETUP DIVERSITY ANALYSES #####</span>
<span class="c1"># !!! SET RAREFACTION DEPTH BASED ON SAMPLE COUNTS !!! #</span>
qiime tools view 1_10_deblur_table.qzv
<span class="nv">rarefaction_depth</span><span class="o">==</span><span class="m">4000</span>
<span class="c1">### SET PATH TO MAPPING FILE ###</span>
<span class="c1"># map_path=map.txt</span>

<span class="c1">### RUN DIVERSITY ANALYSES ###</span>
<span class="nv">diversity_folder</span><span class="o">=</span>1_16_core_metrics_phylogenetic
qiime diversity core-metrics-phylogenetic <span class="se">\</span>
  --i-phylogeny 1_15_deblur_tree.qza <span class="se">\</span>
  --i-table 1_7_deblur_table.qza <span class="se">\</span>
  --p-sampling-depth <span class="nv">$rarefaction_depth</span> <span class="se">\</span>
  --m-metadata-file <span class="s2">&quot;</span><span class="nv">$map_path</span><span class="s2">&quot;</span> <span class="se">\</span>
  --output-dir <span class="s2">&quot;</span><span class="nv">$diversity_folder</span><span class="s2">&quot;</span>

<span class="c1">### VIEW PCoA PLOTS ###</span>
qiime tools view 1_16_core_metrics_phylogenetic/bray_curtis_emperor.qzv
qiime tools view 1_16_core_metrics_phylogenetic/jaccard_emperor.qzv
qiime tools view 1_16_core_metrics_phylogenetic/unweighted_unifrac_emperor.qzv
qiime tools view 1_16_core_metrics_phylogenetic/weighted_unifrac_emperor.qzv

<span class="c1">##### ALPHA DIVERSITY ANALYSES #####</span>
<span class="k">for</span> alpha_metric in faith_pd shannon observed_otus evenness
<span class="k">do</span>
    <span class="nb">echo</span> <span class="s2">&quot;</span><span class="nv">$diversity_folder</span><span class="s2">&quot;</span>/<span class="s2">&quot;</span><span class="nv">$alpha_metric</span><span class="s2">&quot;</span>_vector.qza

    <span class="c1">### ALPHA ASSOCIATION CATEGORICAL MAPPING ###</span>
    qiime diversity alpha-group-significance <span class="se">\</span>
      --i-alpha-diversity <span class="s2">&quot;</span><span class="nv">$divers</span><span class="s2">\ity_folder&quot;</span>/<span class="s2">&quot;</span><span class="nv">$alpha_metric</span><span class="s2">&quot;</span>_vector.qza <span class="se">\</span>
      --m-metadata-file <span class="s2">&quot;</span><span class="nv">$map_path</span><span class="s2">&quot;</span> <span class="se">\</span>
      --o-visualization 1_17_<span class="s2">&quot;</span><span class="nv">$alpha_metric</span><span class="s2">&quot;</span>_group_significance.qzv

    <span class="c1">### ALPHA ASSOCIATION CONTINUOUS MAPPING ###</span>

<span class="k">done</span>

qiime diversity filter-distance-matrix <span class="se">\</span>
  --i-distance-matrix distance-matrix.qza <span class="se">\</span>
  --m-metadata-file samples-to-keep.tsv <span class="se">\</span>
  --o-filtered-distance-matrix identifier-filtered-distance-matrix.qza

<span class="c1">### VIEW RESULTS ###</span>
<span class="c1"># CATEGORICAL #</span>
qiime tools view 1_17_shannon_group_significance.qzv
qiime tools view 1_17_observed_otus_group_significance.qzv
qiime tools view 1_17_evenness_group_significance.qzv
qiime tools view 1_17_faith_pd_group_significance.qzv
</pre></div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="../index.html">Table of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Concepts of Community Analysis</a><ul>
<li><a class="reference internal" href="#quality-control">Quality Control</a></li>
<li><a class="reference internal" href="#community-transformation-and-examination">Community Transformation and Examination</a></li>
<li><a class="reference internal" href="#rarefaction-and-normalization">Rarefaction and Normalization</a></li>
<li><a class="reference internal" href="#alpha-and-beta-diversity">Alpha and Beta Diversity</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="3_sequences_to_composition.html"
                        title="previous chapter">From Sequences to Community Composition</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="5_analysis_and_hypothesis_testing.html"
                        title="next chapter">Microbiome Analysis and Hypothesis Testing</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="../_sources/rst/4_concepts_of_community_analysis.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="5_analysis_and_hypothesis_testing.html" title="Microbiome Analysis and Hypothesis Testing"
             >next</a> |</li>
        <li class="right" >
          <a href="3_sequences_to_composition.html" title="From Sequences to Community Composition"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">VMI Bootcamp II 1.0 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2018, Andrew Brooks, Matt Scholz, Seth Bordenstein.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.8.1.
    </div>
  </body>
</html>